### PR TITLE
bugfix: Prevent main_conf from being modified, in the ngx_http_lua_init_worker function.

### PR DESCRIPTION
**Pull request by wangfakang**
For discussion of related issues, Plz see https://github.com/openresty/lua-nginx-module/issues/1553 .

### CODE REVIEW

**Comment by wangfakang** [comment]
@thibaultcha  Please help to review, thx.

**Comment by thibaultcha** [comment]
@wangfakang I have actually started reviewing this PR. Please allow for some time. My comments are still pending and I may fix them myself locally anyway. Please allow for some time.

**Comment by wangfakang** [comment]
@thibaultcha  Ok, thanks.

**Comment by thibaultcha** [comment]
Good catch. Manually merged with many edits (disregard the leftover comments, I forgot to delete them before approving). Thanks!

File: t/data/fake-merge-module/ngx_http_fake_merge_module.c [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        ngx_conf_log_error(NGX_LOG_ALERT, cf, 0, "create module main conf");
+        return NULL;
+    }
+
+    return fmcf;
+}
+
+
+/* create location configure */
+static void *ngx_http_fake_merge_create_loc_conf(ngx_conf_t *cf)
```
Comment (agentzh) [inline t/data/fake-merge-module/ngx_http_fake_merge_module.c]: These functions do not follow the nginx coding style. We should use `ngx-releng` to check coding style issues in such C source files.

Comment (thibaultcha) [reply t/data/fake-merge-module/ngx_http_fake_merge_module.c]: These were fixed during merge, the merged patch from this PR follows the nginx coding style

**Pull request closed without merging**